contrib/cmus_all.lua

```diff
@@ -0,0 +1,69 @@
+-----------------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2017, JuanKman94 <juan.carlos.menonita@gmail.com>
+-----------------------------------------------------------
+
+-- {{{ Grab environment
+local tonumber = tonumber
+local io = { popen = io.popen }
+local setmetatable = setmetatable
+local string = { gmatch = string.gmatch, format = string.format }
+local helpers = require("vicious.helpers")
+-- }}}
+
+
+-- Cmus: provides CMUS information
+-- vicious.widgets.cmus
+local cmus_all = {}
+
+
+-- {{{ CMUS widget type
+local function worker(format, warg)
+    local cmus_state  = {
+        ["{duration}"]  = 0,
+        ["{file}"] = "N/A",
+        ["{status}"] = "N/A",
+        ["{title}"]  = "N/A",
+        ["{artist}"]  = "N/A",
+        ["{continue}"]  = false,
+        ["{shuffle}"]  = false,
+        ["{repeat}"]  = false,
+    }
+
+    -- Fallback to CMUS defaults
+    local host = warg and (warg.host or warg[1]) or
+		os.getenv("XDG_RUNTIME_DIR") .. "/cmus-socket"
```

Comment:
Where will cmus write to if `XDG_RUNTIME_DIR` is not set?

Reply:
Good observation, tested in a non-XDG environment and it crashed. I'll make a commit